pkg/infra/observer.go

```diff
@@ -60,8 +66,8 @@ func CreateObserver(channel string, node Node, crypto *Crypto, logger *log.Logge
 	return &Observer{Address: node.Addr, d: deliverer, logger: logger}, nil
 }
 
-func (o *Observer) Start(N int, errorCh chan error, finishCh chan struct{}, now time.Time, blockCollector *BlockCollector) {
-	defer close(finishCh)
+func (o *Observer) Start(N int, errorCh chan error, now time.Time, wg *sync.WaitGroup, blockCollector *BlockCollector) {
+	defer wg.Done()
 	o.logger.Debugf("start observer")
 	n := 0
 	for n < N {
```

Comment:
as we discussed earlier, there's no need for `observer` to keep track of total num anymore, right?